app/src/main/kotlin/at/bitfire/davdroid/ui/setup/LoginActivity.kt

```diff
@@ -110,6 +110,8 @@ class LoginActivity @Inject constructor(): AppCompatActivity() {
 
         // start with login info from Intent
         model.updateLoginInfo(loginInfoFromIntent(intent))
+        if (intent.data != null)
+            model.navToPage(LoginScreenModel.Page.LoginDetails)
```

Comment:
OnCreate is called when the activity is re-created, for instance when the screen is rotated. So rotating the screen will always return to the login details page if the Activity was created with an initial Intent.

1. Wrapping the "initial login data" lines into `if (savedInstanceState == null)` should help. Then the code should only be executed on the first creation, but not on re-creation.
2. I think we should check the URI from `LoginInfo` (result of `loginInfoFromIntent`) and not of `intent.data` directly, because it's the task of `loginInfoFromIntent` to convert the Intent to a `LoginInfo` and from then on we don't need the Intent anymore. For instance, if the Intent contains an invalid URI, `LoginInfo` won't contain it and we should not be redirected to the login details page.

Reply:
>     1. Wrapping the "initial login data" lines into `if (savedInstanceState == null)` should help. Then the code should only be executed on the first creation, but not on re-creation.

The code is only run once now. Re-creation / rotation still looses user changes though ...